### PR TITLE
Complete Frogger Game

### PR DESCRIPTION
**Pull request by deveLabR**
### Object-Oriented JavaScript

[Demo](https://develabr.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/deveLabR/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by deveLabR** [comment]
Self-check done.

File: submissions/develabr/frogger/app.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    if (
+      player.x + this.width > this.x &&
+      player.x < this.x + this.width &&
+      player.y + this.height > this.y &&
+      player.y < this.y + this.height
+    ) {
+      player.toStart();
+    }
+  }
```
Comment (OleksiyRudenko) [inline submissions/develabr/frogger/app.js:27]: Requirements posted/linked by bot address two issues with method `update` specifically.
Please identify those, copy/paste them as a comment in this PR and fix code accrodingly.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@deveLabR more than 1 requirements from those posted/linked by bot are not met.

Please copy/paste the requirements, go through them thoroughly, mark those that are met by the code, ask questions here and/or Students' chat if anything is not clear, fix code accordingly.

**Comment by deveLabR** [comment]
If I understood correctly, I need to make corrections according to the following points:

- All numbers like block dimensions, initial locations are defined as constants.
- There are core constants and derived (calculated) constants (e.g. const FIELD_WIDTH = BLOCK_WIDTH * BLOCKS_NUMBER;).
- Separation of Concerns principle is followed (e.g. update method only updates character coordinates and doesn't contain any inline code to check e.g. collisions; calling other methods from update is legitimate).

Did I miss something?

**Comment by deveLabR** [comment]
I hope that I understood what is required of me)) now I will try to fix everything

**Review by OleksiyRudenko — APPROVED** [review]
@deveLabR great job!

**Pull request merged**